docs/style.md

```diff
@@ -125,3 +126,89 @@ While convenient for editing, it may be poorly legible in unrendered form
 <!-- BAD -->
 Definitely don't use soft-wrapping. While markdown mostly ignores source level line breaks, relying on soft wrap makes the source completely unreadable, especially on modern wide displays.
 ```
+
+## [Tracing](https://tracing.rs)
+
+When emitting events and spans with `tracing` prefer adding variable data via
+[`tracing`'s field mechanism][fields].
+
+[fields]: https://docs.rs/tracing/latest/tracing/#recording-fields
+
+Most apparent violation of this rule will be when the message utilizes any form
+of formatting, as seen in the following example:
+
+```rust
+debug!(
+    target: "client",
+    "{:?} Received block {} <- {} at {} from {}, requested: {}",
+    self.client.validator_signer.as_ref().map(|vs| vs.validator_id()),
+    hash,
+    block.header().prev_hash(),
+    block.header().height(),
+    peer_id,
+    was_requested
+);
+```
```

Comment:
Let's use the same style we use elsewhere in the document, with GOOD and BAD markers:

```rust
// GOOD

// BAD
```

Rationale: consistency. It also naively seems that *positive* guidelines, which explain what to do and show good example first, should work better.